.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,29 @@
+<!--
+
+Thanks for wanting to contribute to Numba :)
+
+First, if you need some help or want to chat to the core developers, please
+visit https://gitter.im/numba/numba for real time chat or post to the Numba
+mailing list https://groups.google.com/a/continuum.io/forum/#!forum/numba-users.
+
+Here's some guidelines to help the review process go smoothly.
+
+0. Please write a description in this text box of the changes that are being
+   made.
+
+1. Please ensure that you have written units tests for the changes made/features
+   added.
+
+2. If you are closing an issue please use one of the automatic closing words as
+   noted here: https://help.github.com/articles/closing-issues-using-keywords/
+
+3. If your pull request is not ready for review but you want to make use of the
+   continuous integration testing facilities please label it with `[WIP]` and
+   then remove the label when you'd like it to be reviewed.
+
+4. Once review has taken place please do not add features or make changes out of
```

Comment:
I wonder if we should mention that user should avoid rebasing once a review has taken place, because rebasing would erase the GH review history.

Reply:
Good idea, will add as `5.`. Thanks.

Reply:
Good idea.